### DISCUSSION TITLE
How to upload to arweave at backend using solana metaplex js sdk?

### DISCUSSION THREAD

**Discussion by JohnVersus** [metaplex-foundation/js#293] [Q&A] [answered]
I have tried uploading metadata to arweave from the frontend using the `metaplex.nfts().uploadMetadata` and it is working great. I would like to know if it is possible to upload from the backend using `metaplex.nfts().uploadMetadata` and privateKey. If possible please share an example code or docs for the same. 

Thanks🙏

**Comment by JohnVersus** [comment, marked as answer]
I figured this out. Here is the solution in case if some have the same question
```
const connection = new Connection(clusterApiUrl('devnet'));
  const metaplex = new Metaplex(connection);
  const key = Uint8Array.from(
    base58.decode('PrivateKey'),
  );

  const keypair = Keypair.fromSecretKey(key);
  console.log(`keypair created: ${keypair.publicKey.toString()}`);

  metaplex.use(walletAdapterIdentity(keypair));
  metaplex.use(
    bundlrStorage({
      address: 'https://devnet.bundlr.network',
      providerUrl: 'https://api.devnet.solana.com',
      timeout: 60000,
      identity: keypair,
    }),
  );
```

**Comment by devusvulgaris** [comment]
Is there a way to send private key from Phantom wallet from frontend to backend. In case I want to upload file or create NFT on backend? I would need private key to recreate keypair on backend.

**Reply by JohnVersus** [reply]
That does not seem like a good idea to send privatekey from frontend. You need to have the private stored on the backend env file, for safety reasons. 

```js
// use this to generate the solana key pair in backend
const key = Uint8Array.from(base58.decode(process.env.PRIVATE_KEY));
const keypair = Keypair.fromSecretKey(key);
```
